### ISSUE TITLE
.app crashes when run from finder, running unix executable "./myapp" works fine.

### ISSUE THREAD

**Issue by tomtatchell** [pyinstaller/pyinstaller#2597]
Hello,

I've tried googling to find solutions to the problem with no real luck.
I thought I might have been onto something when I found a post about the application's working directory being non-writable. But I couldn't seem to get that working for me even after setting it via chdir(). [Stackoverflow link with the same problem](http://stackoverflow.com/questions/26176169/pyinstaller-os-x-app-runs-from-command-line-but-not-finder-window)  
  
## A little about the script  
The script creates a gui using PyQt5. Within the PyQt5, there is a progress bar and drag-drop action, a button push (escape to exit) and lets the user drag the interface around by clicking and dragging anywhere.
The idea was the user would drag a folder containing an image sequence onto the application and it would automatically convert it by running a shell command to ffmpeg getting it to convert the sequence to a .mov file.
The application only crashes when I drag and drop the folder. I can use other functions fine (escape key to exit, drag window around from anywhere).
If it helps, you can find a copy of the script [here](https://github.com/tomtatchell/img-seq-converter/blob/master/img_seq_convert.py).

## My Problem  
  
Double clicking on the created .app file (img_seq_convert.app) opens it fine, then crashes on user input. However, running the unix executable from within img_seq_convert.app works fine, as does the unix executable created from `--onefile`.

#### pyinstaller --debug console output
So I run `pyinstaller img_seq_convert.py --onefile --windowed --debug` to bundle the files, I get the following output: [ bundle_debug_console_output.txt ](https://gist.github.com/tomtatchell/5d5538c4c634e2989daddfac52b3703a)  
  
#### console output from running ./dist/img_seq_convert  
Upon running `./dist/img_seq_convert` (the unix executable created by pyinstaller) It runs absolutely fine and behaves as I'd expect it to.
[Here is the console output when executed.](https://gist.github.com/tomtatchell/4627510e486e19cfdbef379381679d05)
I get exactly the same output if I were to cd into the app `cd img_seq_convert.app/Contents/MacOS` and run `./img_seq_convert`
  
#### Console (not terminal) output during double click and crash  
I opened the console utility and cleared the screen. Ran the application through until the crash (user input) and have captured the output.  
[So this is what the console is saying as I run and crash the img_seq_convert.app](https://gist.github.com/tomtatchell/ae0fec2698a748c536f7886572e1e608)  
[And this is the crash report that was generated](https://gist.github.com/tomtatchell/bf39c3d47880267adf33e3fedfeb42a9)  
  
## General Environment Info  
I'm running Python 3.5.3 and pyinstaller 3.2.1  (installed via `pip3 install pyinstaller`)
The modules I'm importing to my script are:
sys  
from os: path, getcwd, chdir  
[pyseq](https://github.com/rsgalloway/pyseq)  
[pexpect](https://pexpect.readthedocs.io/en/stable/)  
from PyQt5.QtWidgets: QWidget, QMainWindow, QProgressBar, qApp, QMessageBox, QApplication, QLabel  
from PyQt5.QtGui: QPalette, QImage, QBrush  
from PyQt5.QtCore: Qt, QSize  
  
I'm running all of this on MacOSX 10.11.6 El Capitan and editing the code through PyCharm CE.
  
  
Thanks in advance!!!  

Any suggestions / help would be awesome!  
  
Cheers  
  
Tom

**Comment by derchrisuk** [comment]
Funny, we are actually facing a very similar issue.
PyInstaller is working fine, and we end up with an .app file.
However, running this from the Desktop or CLI ends with the following error:

`LSOpenURLsWithRole() failed for the application <path/to/.app> with error -10810`

When browsing the .app content into `Contents/MacOS`, and running the executable works fine.
So does creating a one file version of it using `--onfile`

**Comment by AlexKucera** [comment]
I have the same problem.

```LSOpenURLsWithRole() failed for the application <path/to/.app> with error -10810```

And if I build using `--onefile` I get a working executable as well as an app bundle that "looks" right. Both the executable and the exe inside the app bundle works perfectly when double clicked. But trying to start the app bundle doesn't work.

In fact, trying the open the app bundle with `open <pathto/.app>` does nothing, not even throw an error. And no Console output either.

**Comment by Kelketek** [comment]
I am also having this problem. Running the executable inside my .app directory and the executable built stand-alone both work without issue, but when I use `open -a`, I get:

```
LSOpenURLsWithRole() failed for the application /Path/to/package.app with error -10810.
```

Is it possibly something to do with the Info.plist?

**Comment by Dlangowski** [comment]
Did anyone figure this out? I'm experiencing the same problem

**Comment by derchrisuk** [comment]
Nope, and we have abandoned pyinstaller in favor of a shell script to run the app

**Comment by Johannes0Horn** [comment]
Anyone any idea? I am facing the same Problem for moviepy
